### ISSUE TITLE
pika can't encode unsigned long long header values that don't fit in long long

### ISSUE THREAD

**Issue by elelayan** [pika/pika#1265]
I try to send a huge number in an application property header (11171330917854890490 which is between 2^63 and 2^64) but it fails with and exception:

(using Python 3.6.7 | packaged by conda-forge | (default, Nov 21 2018, 03:09:43) )

Stacktace in context is:

```
    self._channel.basic_publish(self._exchange, self._routing_key, self._message, properties)
  File "/.../lib/python3.6/site-packages/pika/channel.py", line 428, in basic_publish
    mandatory=mandatory), (properties, body))
  File "/.../lib/python3.6/site-packages/pika/channel.py", line 1401, in _send_method
    self.connection._send_method(self.channel_number, method, content)
  File "/.../lib/python3.6/site-packages/pika/connection.py", line 2232, in _send_method
    self._send_message(channel_number, method, content)
  File "/.../lib/python3.6/site-packages/pika/connection.py", line 2253, in _send_message
    marshaled_body_frames.append(frame_header.marshal())
  File "/.../lib/python3.6/site-packages/pika/frame.py", line 103, in marshal
    pieces = self.properties.encode()
  File "/.../lib/python3.6/site-packages/pika/spec.py", line 2179, in encode
    data.encode_table(pieces, self.headers)
  File "/.../lib/python3.6/site-packages/pika/data.py", line 93, in encode_table
    tablesize += encode_value(pieces, value)
  File "/.../lib/python3.6/site-packages/pika/data.py", line 133, in encode_value
    pieces.append(struct.pack('>cq', b'l', value))
struct.error: int too large to convert
```

To reproduce, I isolated this short code in the python interpreter:

imports:
```
>>> import pika.compat
>>> import pika.data
```

verify int size
```
>>> 11171330917854890490 < 2**64
True
>>> 11171330917854890490 < 2**63
False
```

try converting a small long: it works
```
>>> a=[];pika.data.encode_value(a, pika.compat.long(10));a
9
[b'l\x00\x00\x00\x00\x00\x00\x00\n']
```

try converting a big long: it fails
```
>>> a=[];pika.data.encode_value(a, pika.compat.long(11171330917854890490));a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/.../lib/python3.6/site-packages/pika/data.py", line 132, in encode_value
    pieces.append(struct.pack('>cq', b'l', value))
struct.error: int too large to convert
```

AFAICT (pdf page 31) the amqp 0-9-1 format specifier `b'l'` is for long-long-uint, which is correct to fit this number greater than 2^63.

But the struct.pack [format string](https://docs.python.org/3.8/library/struct.html#format-characters) is incorrect: it should be **'>cQ'** (upper-case Q).

Success with **'>cQ'** specifier
```
>>> import struct
>>> struct.pack('>cQ', b'l', 11171330917854890490)
b'l\x9b\x08\x8aH\x13\x14I\xfa'
```

Failure with **'>cq'** specifier
```
>>> struct.pack('>cq', b'l', 11171330917854890490)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
struct.error: int too large to convert
```

**Comment by elelayan** [comment]
Sure, that could be a problem if negative long longs are packed this way. Maybe check if < 0 ?

**Comment by lukebakken** [comment]
Hello and thanks for using Pika. You seem to have a good grasp on this issue. Please submit a pull request to fix it, and include tests.

**Comment by lukebakken** [comment]
I removed this from the [`1.2.0`](https://github.com/pika/pika/milestone/17) milestone as it is not a serious enough issue to address at this point. @elelayan if you have time to submit a pull request that would be great.

**Comment by michaelklishin** [comment]
@elelayan would you be interested in submitting a PR for your suggestion?

**Comment by elelayan** [comment]
Hi, @michaelklishin. Unfortunately I don't have time to implement and test sufficiently right now.

**Comment by lukebakken** [comment]
Fixed by #1345
